infinity/stream.py

```diff
@@ -510,6 +511,43 @@ def load_object (self):
     def __repr__ (self):
         return "<ResourceStream: %s at 0x%08x>" %(self.resref, id (self))
 
+class OverrideStream (MemoryStream):
+    """A subclass of MemoryStream for looking up data in the override directory."""
+
+    def __init__ (self):
+        MemoryStream.__init__ (self)
+
+    def open (self, name, filetype = None, index = 0):
+        self.resref = name
+        self.type = core.ext_to_type(filetype)
+
+        if not core.override:
+            raise RuntimeError, "Override is empty."
+
+        obj = core.search_override (self.resref, self.type)
+
+        if len (obj) > 1 and self.type is None:
+            raise RuntimeError, name + ": more than one result in override, types " + ' ' . join([ str(o['type']) for o in obj ])
+
+        if len (obj) == 0:
+            raise RuntimeError, name + ": not found in override!"
+
+        try:
+            obj = obj[index]
+        except IndexError:
+            obj = obj[0]
```

Comment:
I can't imagine where passing the index would be needed, but still, silently using a fallback doesn't feel nice.